base/compiler/ssair/ir.jl

```diff
@@ -780,8 +780,10 @@ function renumber_ssa2(val::SSAValue, ssanums::Vector{Any}, used_ssa::Vector{Int
     if do_rename_ssa
         val = ssanums[id]
     end
-    if isa(val, SSAValue) && used_ssa !== nothing
```

Comment:
Is there a performance advantage to avoiding the `&&` here?

Reply:
yeah, inference can't see through `&&`, so it's best not to combine `isa` with anything else in the same expression